### ISSUE TITLE
Improve console output

### ISSUE THREAD

**Issue by PizieDust** [robur-coop/mollymawk#37]
Currently, unikernel console output is gotten by constantly polling the server using javascript loops. While this works, it's not very ideal. 
A websocket will be much more intuitive for such a purpose. 
A good start for this task will be to properly research the state of websockets as concerns web in OCaml. 
Figure out if the currently available websocket packages/libraries can be used/ reliable and if not, investigate if it's worthwhile writing a websocket library.

**Comment by hannesm** [comment]
What I find pretty crucial is that at the moment whenever someone looks into the "unikernel info", the JavaScript starts to read the console output peridically (-> lots of HTTP requests), and mollymawk starts an albatross connection to read the console output (and store it in mollymawk).

The albatross connection reading the console output is never terminated, and that is wasteful. In addition, in mollymawk we have then for each unikernel console a Lwt task that reads and pushes into a "ring buffer".

A different approach may be to say "you can read the last 20 lines of the console", but do not do this asynchronously. That may be sufficient for the use cases (you then can reload the page to see more up-to-date information). It would as well work around the websocket question, and reduce the overall complexity.

What do you think?

**Comment by hannesm** [comment]
See #50 for a non-polling version, no websockets (yet).

**Comment by hannesm** [comment]
With #50, there's less pressure on albatross and on the client side -- but we barely see more than two lines of console output.

We should find a better way on how to achieve this.... maybe it is really digging into websockets...

**Comment by PizieDust** [comment]
Indeed. I hope to work on websockets at a point when we have cleared most of the issues currently open.
